app/src/org/commcare/android/logic/BarcodeScanListenerDefaultImpl.java

```diff
@@ -0,0 +1,121 @@
+package org.commcare.android.logic;
+
+import android.app.Activity;
+import android.content.ActivityNotFoundException;
+import android.content.Intent;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import org.commcare.dalvik.BuildConfig;
+import org.commcare.suite.model.Callout;
+import org.commcare.suite.model.CalloutData;
+
+import java.util.Hashtable;
+import java.util.Map;
+
+/**
+ * Created by dancluna on 8/5/15.
+ */
+public final class BarcodeScanListenerDefaultImpl {
+    public interface BarcodeScanListener {
+        void onBarcodeFetch(String result, Intent intent);
+
+        void onCalloutResult(String result, Intent intent);
+    }
```

Comment:
I like the idea of moving this interface into its own file. This will make all the uses of it much cleaner, because it won't have to be prefixed by BarcodeScanListenerDefaultImpl